akka-docs/src/main/paradox/typed/persistence-active-active.md

```diff
@@ -0,0 +1,191 @@
+ 
+![images/lww.png](images/lww.png)
+
+There is a small utility class @api[LwwTime] that can be useful for implementing last writer wins semantics.
+It contains a timestamp representing current time when the event was persisted and an identifier of the
+data center that persisted it. When comparing two @api[LwwTime] the greatest timestamp wins. The replica
+identifier is used if the two timestamps are equal, and then the one from the data center sorted first in
+alphanumeric order wins.
+
+The nature of last writer wins means that if you only have one timestamp for the state the events must represent an
+update of the full state, otherwise there is a risk that the state in different data centers will be different and
+not eventually converge to the same state.
+
+An example of that would be an entity representing a blog post and the fields `author` and `title` could be updated
+separately with events @scala[`AuthorChanged(newAuthor: String)`]@java[`new AuthorChanged(newAuthor)`] and @scala[`TitleChanged(newTitle: String)`]@java[`new TitleChanged(newTitle)`].
+
+Let's say the blog post is created and the initial state of `title=Akka, author=unknown` is in sync in both data centers `DC-A` and `DC-B`.
+
+In `DC-A` author is changed to "Bob" at time `100`. Before that event has been replicated over to `DC-B` the
+title is updated to "Akka News" at time `101` in `DC-B`. When the events have been replicated the result will be:
+
+`DC-A`: The title update is later so the event is used and new state is `title=Akka News, author=Bob`
+
+`DC-B`: The author update is earlier so the event is discarded and state is `title=Akka News, author=unknown`
+
+The problem here is that the partial update of the state is not applied on both sides, so the states have diverged and will not become the same.
+
+To solve this with last writer wins the events must carry the full state, such as @scala[`AuthorChanged(newContent: PostContent)`]@java[`new AuthorChanged(newContent)`] and @scala[`TitleChanged(newContent: PostContent)`]@java[`new TitleChanged(newContent)`]. Then the result would eventually be `title=Akka News, author=unknown` on both sides.
+The author update is lost but that is because the changes were performed concurrently. More important is that the state
+is eventually consistent.
+
+Including the full state in each event is often not desired. An event typically represent a change, a delta. Then one can use several timestamps, one for each set of fields that can be updated together. In the above example one could use one timestamp for the title and another for the author. Then the events could represent changes to parts of the full state, such as @scala[`AuthorChanged(newAuthor: String)`]@java[`new AuthorChanged(newAuthor)`] and @scala[`TitleChanged(newTitle: String)`]@java[`new TitleChanged(newTitle)`].
```

Comment:
```suggestion
Including the full state in each event is often not desired. An event typically represent a change, a delta. Then one can use several timestamps, one for each set of fields that can be updated together. In the above example one could use one timestamp for the title and another for the author. Then the events could represent changes to deltas of the full state, such as @scala[`AuthorChanged(newAuthor: String)`]@java[`new AuthorChanged(newAuthor)`] and @scala[`TitleChanged(newTitle: String)`]@java[`new TitleChanged(newTitle)`].
```

Better to use the word `delta`. It the common term and it mentioned already before.

Diff is not showing the suggestion. It's in "Then the events could represent changes to deltas of the full state, such as .."

Reply:
I personally found the blog example not great for a LWW. At least, not in the way we are using it.

We show that we will lose updates to them suggest that we should put the whole state in the event. Which again causes data loss, but at least guarantees eventual consistency. We finally mention that it's a bad idea to carry the whole state in the events (which is completely true).

Timestamp per field is a good approach and that's probably how people should do it in a blog post with LWW.

Reply:
"changes to the delta" doesn't that mean changes to the changes?

Reply:
I don't think the example is too important here, it is to get across the risks of LWW.

Reply:
> "changes to the delta" doesn't that mean changes to the changes?

Correct, I will adapt the suggestion.

> I don't think the example is too important here, it is to get across the risks of LWW.

True, I see now that the goal of the example is to illustrate the risk.

Reply:
is it better if we say:

"Then the events could represent the deltas of the full state..."